python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -416,6 +417,15 @@ def partitions_def(self) -> Optional[PartitionsDefinition]:
         """
         return None if not self.partitioned_config else self.partitioned_config.partitions_def
 
+    @property
+    def backfill_policy(self) -> Optional[BackfillPolicy]:
+        first_key = next(iter(self.asset_layer.executable_asset_keys), None)
+        if self.partitions_def is None or first_key is None:
+            return None
+        # It is enforced that all assets in a job have the same backfill policy, so we can just take
+        # the first backfill policy.
```

Comment:
Consider an invariant to enforce this as a sanity check and using `@cached_property`